app/src/main/kotlin/hr/asee/android/template/compose/ui/prelogin/register/RegisterViewModel.kt

```diff
@@ -0,0 +1,169 @@
+
+    private fun onEmailTextChange(newValue: String) {
+        _emailState.update { it.copy(text = newValue, isError = false) }
+    }
+
+    private fun onPasswordTextChange(newValue: String) {
+        _passwordState.update { it.copy(text = newValue, isError = false) }
+    }
+
+    private fun onConfirmPasswordTextChange(newValue: String) {
+        _confirmPasswordState.update { it.copy(text = newValue, isError = false) }
+    }
+
+    fun register() {
+        showLoading()
+        runSuspend { registerInternal() }
+    }
+
+    private suspend fun registerInternal() {
+        registerUseCase(RegisterUseCase.RegisterRequest(name = nameState.value.text, email = emailState.value.text, password = emailState.value.text, confirmpassword = confirmPasswordState.value.text)).onFinished(
+            successCallback = this::onRegisterSuccessful,
+            errorCallback = this::onRegisterError,
+        )
+    }
+
+    private fun onRegisterSuccessful() {
+        router.navigateToPostLoginScreen()
+    }
+
+    private fun onRegisterError(errorData: ErrorData) {
+        when (errorData.errorType) {
+            RegisterUseCase.RegisterError.MISSING_PASSWORD_ERROR -> showError(RegisterMessages.MISSING_PASSWORD_ERROR)
+            RegisterUseCase.RegisterError.STORE_ACCESS_TOKEN_ERROR -> showError(RegisterMessages.STORE_ACCESS_TOKEN_ERROR)
+            RegisterUseCase.RegisterError.USER_NOT_FOUND_ERROR -> showError(RegisterMessages.USER_NOT_FOUND_ERROR)
+            RegisterUseCase.RegisterError.MISSING_EMAIL_OR_USERNAME_ERROR -> showError(RegisterMessages.MISSING_EMAIL_OR_USERNAME_ERROR)
+            else -> showError(CommonMessages.UNEXPECTED_ERROR)
+        }
+    }
+
+    fun goToLogin() {
+        router.navigateToLoginScreen()
+    }
+
+    fun isEmailValid() : Boolean{
+        if (emailState.value.text.isNotEmpty() && !hasValidEmailDomain(emailState.value.text)){
+            return true
+        }
+        return false
+    }
+
+    fun isPasswordValid() : Boolean{
+        if (passwordState.value.text.isNotEmpty() && !isValidPasswordFormat(passwordState.value.text) && passwordState.value.text.length >= 8) {
+            return true
+        }
+        return false
+    }
+
+    fun isPasswordLongEnough() : Boolean{
+        if(passwordState.value.text.isNotEmpty() && passwordState.value.text.length < 8){
+            return true
+        }
+        return false
+    }
+
+    fun isValidConfirmPassword() : Boolean{
+        if (!isFlagUp() && confirmPasswordState.value.text.isNotEmpty()){
+            return true
+        }
+        return false
+    }
+    fun isFlagUp() : Boolean{
+        if (isValidPasswordFormat(passwordState.value.text) && passwordState.value.text.length >= 8) {
+            if (hasValidEmailDomain(emailState.value.text)) {
+                if (passwordState.value.text == confirmPasswordState.value.text) {
+                    return true
+                }
+            }
+        }
+        return false
+    }
```

Comment:
Ovako nekako je trebalo, samo što UI, odnosno `RegisterScreen` u ovom slučaju ne bi smio dobiti vrijednost od view modela, to je jednosmjerna komunikacija, oni komuniciraju preko state polja koje UI promatra i onda ovisno o tome mijenja izgled ekrana ili prikaže poruku. Svi ovi stateovi gore (`_nameState`, `_emailState` itd. imaju varijablu `isError` koju možeš promijeniti u view modelu, a slušati u `RegsiterScreen` ekranu. Ove metode za provjeru isValid itd. možeš pozivati u ovim textChange metodama gore, jer će se pozvati svaki put kad se unese tekst.